repos/c2sm/packages/py-icon4py/package.py

```diff
@@ -33,8 +33,16 @@ class PyIcon4py(PythonPackage):
     depends_on('py-tabulate@0.8.9:', type=('build', 'run'))
     # TODO: push new version to Spack official
     depends_on('py-fprettify@0.3.7:', type=('build', 'run'))
+    depends_on('py-cffi@1.5.0:', type=('build', 'run'))
+    depends_on('py-netcdf4', type=('build', 'run'))
+    depends_on('py-mpi4py@3.0:', type=('build', 'run'))
+    depends_on('py-pytz', type=('build', 'run'))
+    depends_on('py-ghex@0.3.1', type=('build', 'run'))
+    depends_on('py-wget', type=('build', 'run'))
+    depends_on('serialbox@2.6.1_2023-06-12 +python', type=('build', 'run'))
```

Comment:
The new dependencies are only needed for recent versions of py-icon4py.
Add `when=@x.x.x:` to decaration.